baggage/baggage_test.go

```diff
@@ -339,6 +339,13 @@ func TestBaggageParse(t *testing.T) {
 				"foo": {Value: "2"},
 			},
 		},
+		{
+			name: "url encoded value",
+			in:   "key1=val%252",
```

Comment:
This test was insufficient. The regexp check now happens _after_ `QueryUnescape`, not at all what was intended in the original code. As a result, valid strings like `val%27` do not pass the check. This specific test does pass because the decoded string `val%2` still "looks like" URL-encoded (i.e. it passes the regex).

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/3685#issuecomment-1421989032